lib/common/xml.c

```diff
@@ -2172,8 +2172,19 @@ pcmk__xml_cleanup(void)
     xmlCleanupParser();
 }
 
+/*!
+ * \internal
+ * \brief Get the XML element whose \c PCMK_XA_ID matches an \c PCMK_XA_ID_REF
+ *
+ * \param[in] input   Element whose \c PCMK_XA_ID_REF attribute to check
+ * \param[in] search  Root of search for node with matching \c PCMK_XA_ID
+ *                    (\c NULL to use \p input)
+ *
+ * \return If \p input has a \c PCMK_XA_ID_REF attribute, node in \p search
+ *         whose \c PCMK_XA_ID attribute matches; otherwise, \p input
+ */
 xmlNode *
-expand_idref(xmlNode * input, xmlNode * top)
+pcmk__xe_expand_idref(xmlNode *input, xmlNode *search)
```

Comment:
If we can get `expanded_xml` into the CIB (`scheduler->input`) safely, so that it's in the same document, then we might be able to drop the `search` argument after tracing through the other paths. Not worth it for this PR but it's on the table. That might turn out too messy with digests and output and such.